impl/client/src/main/javascript/web/pentaho/visual/base/view.js

```diff
@@ -0,0 +1,1322 @@
+         *
+         * @see pentaho.visual.base.View.Type#extension
+         */
+        get extensionEffective() {
+          var effective = this._extensionEf;
+          if(effective === undefined) {
+            effective = null;
+
+            var ancestor = this.ancestor;
+            if(ancestor.isSubtypeOf(View.type)) {
+              var ancestorExtEf = ancestor.extensionEffective;
+              if(ancestorExtEf) {
+                effective = {};
+                specUtil.merge(effective, ancestorExtEf);
+              }
+            }
+
+            if(this._extension) {
+              if(!effective) effective = {};
+              specUtil.merge(effective, this._extension);
+            }
+
+            this._extensionEf = effective;
+          }
+
+          return effective;
+        }
+        // endregion
+      }
+    }, /** @lends pentaho.visual.base.View */{
+
+      // TODO: convert to use a view spec?
+      // region static factory sugar
+      /**
+       * Creates a view, asynchronously, given its specification.
+       *
+       * If the view specification has its type annotated inline,
+       * a view of that type is built.
+       *
+       * Otherwise, when the view type is not annotated inline in the specification,
+       * if its `model` property is specified with an already instantiated model instance,
+       * or with a type annotated specification,
+       * that model's type's default view class is used to create a view instance from
+       * the given specification.
+       * The returned promise is rejected if the provided model specification is
+       * not specified or is not type annotated.
+       * The returned promise is also rejected if the model type has no registered default view type.
+       *
+       * Unlike the {@link pentaho.type.Type#createAsync} counterpart method,
+       * this static variant can be called to create an instance of any view type,
+       * even if it isn't a subtype of `this` one.
+       *
+       * @param {pentaho.visual.base.spec.IViewEx} viewSpec - The extended view specification.
+       *
+       * @return {!Promise.<pentaho.visual.base.View>} A promise for a view with the given specification.
+       *
+       * @rejects {pentaho.lang.ArgumentRequiredError} When `viewSpec` is not specified.
+       * @rejects {pentaho.lang.ArgumentRequiredError} When `viewSpec` has no annotated type, inline, and
+       * the `model` property is unspecified.
+       * @rejects {pentaho.lang.ArgumentRequiredError} When `viewSpec` has no annotated type, inline, and
+       * the `model` property is a specification which also does not have its type annotated inline.
+       * @rejects {Error} When there isn't a registered default view class for the type of `model`.
+       * @rejects {Error} When the registered default view class does not exist or otherwise fails to load
+       * by {@link pentaho.type.Context#getAsync}.
+       */
+      createAsync: function(viewSpec) {
```

Comment:
When methods are either sync or async, that's preferable, I guess. However, because we have `pentaho.type.Type#create` and `pentaho.type.Type#createAsync`, `pentaho.type.Context#get` and `pentaho.type.Context#getAsync`, and `pentaho.type.Context#getAllAsync`, side-by-side, it would be inconsistent to have an asynchronous`pentaho.visual.base.View.create` method that just slightly changes the semantics of  `pentaho.type.Type#createAsync`.